### ISSUE TITLE
pandoc: openFile: does not exist (No such file or directory)

### ISSUE THREAD

**Issue by harryprince** [rstudio/rmarkdown#1268]
I am trying to server Rmd using shiny server. However, it seems rmarkdown or pandoc is buggy.

sessionInfo():

```
R version 3.3.2 (2016-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 14.04.5 LTS

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base
```

here is the shiny server log:

```
processing file: index.Rmd
output file: /tmp/rmarkdown/e01d3e5bbe48ef364e37eb0580046563/index.knit.md

pandoc: /tmp/RtmpWH829n/rmarkdown-str67a94810a85d.html: openFile: does not exist (No such file or directory)
Warning: Error in : pandoc document conversion failed with error 1
Stack trace (innermost first):
    115: pandoc_convert
    114: convert
    113: <Anonymous>
    112: do.call
    111: contextFunc
    110: .getReactiveEnvironment()$runWith
    109: shiny::maskReactiveContext
    108: <reactive>
     95: doc
     94: shiny::renderUI
     93: func
     92: force
     91: withVisible
     90: withCallingHandlers
     89: globals$domain$wrapSync
     88: promises::with_promise_domain
     87: captureStackTraces
     83: tryCatch
     82: do
     81: hybrid_chain
     80: origRenderFunc
     79: output$__reactivedoc__
      3: <Anonymous>
      2: do.call
      1: rmarkdown::run
```

**Comment by yihui** [comment]
Please test the development version:

```r
devtools::install_github('rstudio/rmarkdown')
```

https://github.com/rstudio/blogdown/issues/256#issuecomment-365966430

**Comment by harryprince** [comment]
I tried to do it , however, it failed too and the error is the same as usual.

```
 devtools::install_github("rstudio/rmarkdown",lib="/usr/local/lib/R/site-library")
```

**Comment by yihui** [comment]
Are you sure R uses the rmarkdown package from `/usr/local/lib/R/site-library`? In other words, have you checked `.libPaths()` and are you sure `/usr/local/lib/R/site-library` is the first entry? When showing the session info, it will be great if you can load the package before showing it, e.g., either

```r
library(rmarkdown)
sessionInfo()
```

or better (as requested in the [issue template](https://github.com/rstudio/rmarkdown/blob/master/.github/ISSUE_TEMPLATE.md)):

```r
devtools::session_info('rmarkdown')
```

**Comment by ADraginda** [comment]
Just FYI I'm also seeing this error (trying to use rmarkdown with flexdashboard). Using the dev version unblocked things but I'm still seeing the error.

**Comment by yihui** [comment]
@ADraginda Are you also on Shiny Server? Is it reproducible locally?

**Comment by NilsOle** [comment]
I have the same issue, but only when I run rmarkdown::render(...) in parallel on multiple cores. When I do 
`registerDoParallel(cores=1)`
it works without error. When running in parallel, out of many calls of the function, only some return errors as described here, and also not consistently (appears random at first sight).
Is that information helpful?

The code to reproduce the issue can be found here: https://gist.github.com/hrbrmstr/17bc21af55392f23f012f57bb2fda51c

**Comment by goldingn** [comment]
I can reproduce this error locally. Using `rmarkdown::pandoc_convert()` to convert a file in the same directory works, but trying to convert it one directory down fails.

Create directory structure and file
``` r
dir <- tempdir()
sub_dir <- tempfile(tmpdir = dir)
f <- tempfile(tmpdir = sub_dir)
dir.create(sub_dir)
writeLines("This is a test", f)
cwd <- getwd()
library(rmarkdown)
```

This works

``` r
setwd(sub_dir)
fname <- basename(f)
readLines(fname)
#> [1] "This is a test"
pandoc_convert(fname, to = "html")
```

This doesn't:

``` r
setwd(dir)
fname <- file.path(basename(sub_dir), basename(f))
readLines(fname)
#> [1] "This is a test"
pandoc_convert(fname, to = "html")
#> Error: pandoc document conversion failed with error 1
```

``` r
setwd(cwd)
sessionInfo()
#> R version 3.5.1 (2018-07-02)
#> Platform: x86_64-apple-darwin15.6.0 (64-bit)
#> Running under: macOS Sierra 10.12.6
#> 
#> Matrix products: default
#> BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
#> LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
#> 
#> locale:
#> [1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] rmarkdown_1.10.12
#> 
#> loaded via a namespace (and not attached):
#>  [1] compiler_3.5.1  magrittr_1.5    htmltools_0.3.6 tools_3.5.1    
#>  [5] yaml_2.2.0      Rcpp_0.12.18    stringi_1.2.4   knitr_1.20     
#>  [9] htmldeps_0.1.1  digest_0.6.16   stringr_1.3.1   evaluate_0.11
```

**Comment by AndrewManHayChiu** [comment]
I had a similar problem, where I could knit in some directories, but not in others (on a mapped network drive).

I removed pandoc 2.4, to direct RStudio to use the in-built pandoc as per @yihui 's comment at #1184.

**Comment by pauvasquezh** [comment]
@NilsOle I'm having the same problem as you are describing, were you able to solve it somehow? :)

**Comment by vthivierge** [comment]
Following @pauvasquezh, @NilsOle or @goldingn were you able to solve this?

**Comment by vthivierge** [comment]
> Following @pauvasquezh, @NilsOle or @goldingn were you able to solve this?

I solved it by changing my RStudio global options and setting the R version to be the local C:\ drive and not R on the network drive.

**Comment by crossxwill** [comment]
If the Rmd file is stored on a network drive, the knitting fails. However, if the Rmd file is stored on the local C:\ drive, the knitting works fine.

## Network drive
```
output file: ppt_test.knit.md

"C:/Users/wchiu/AppData/Local/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS ppt_test.utf8.md --to pptx --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash+smart --output ppt_test.pptx 
pandoc.exe: ppt_test.utf8.md: openBinaryFile: does not exist (No such file or directory)
Error: pandoc document conversion failed with error 1
Execution halted
```

## Local C:\ drive
```
"C:/Users/wchiu/AppData/Local/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS ppt_test.utf8.md --to pptx --from markdown+autolink_bare_uris+ascii_identifiers+tex_math_single_backslash+smart --output ppt_test.pptx 
output file: ppt_test.knit.md


Output created: ppt_test.pptx
```

## Session Info
```
> sessionInfo()
R version 3.6.1 (2019-07-05)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] compiler_3.6.1  htmltools_0.3.6 tools_3.6.1     yaml_2.2.0      Rcpp_1.0.1      rmarkdown_1.14 
 [7] knitr_1.23      xfun_0.8        digest_0.6.20   evaluate_0.14  
```

## RStudio Version
```
$citation

To cite RStudio in publications use:

  RStudio Team (2018). RStudio: Integrated Development for R. RStudio, Inc., Boston, MA URL
  http://www.rstudio.com/.

A BibTeX entry for LaTeX users is

  @Manual{,
    title = {RStudio: Integrated Development Environment for R},
    author = {{RStudio Team}},
    organization = {RStudio, Inc.},
    address = {Boston, MA},
    year = {2018},
    url = {http://www.rstudio.com/},
  }


$mode
[1] "desktop"

$version
[1] ‘1.2.1335’
```

**Comment by lamontbeau** [comment]
Just save the Rmd file to the C:\ drive rather than desktop (Network Drive). This solved my problem while I have struggled a month...
Very simple solution. Thank you guys.

**Comment by crossxwill** [comment]
@yihui  - although there is a very simple work around (store Rmd file on C:\ rather than network drive), some of us are not allowed to save files on our C:\ (per company policy). Is there a plan to fix rmarkdown or knitr such that Rmd files thare stored on a network drive can still knit?

**Comment by cderv** [comment]
I think this is the same issue as in #701 where there is already a long discussion. 

An issue was even opened to pandoc (jgm/pandoc#1326) as it is in fact a more general issue with pandoc than just Rmarkdown. it was conclude that it could come from the drive directly, and that it is not specific to R or Rmarkdown tools (https://github.com/rstudio/rmarkdown/issues/701#issuecomment-236960677)

Currently solution is to copy locally **somewhere** (not necessary in C:) before doing the conversion. see for example https://github.com/rstudio/rmarkdown/issues/701#issuecomment-237222070 or other comments. 

To share experience, on a RStudio Server Pro (linux) with NFS mounted drive, we have no issue with that. 
Before I had issues with older system, and specifically with `self_contained=TRUE` document.
Also, there was recently a specific fix for WINDOWS environment in https://github.com/jgm/pandoc/issues/5127 that is included in Pandoc 2.7.3. I don't know if it could help. 

At the time, I manage to change some directory argument from Rmarkdown (https://github.com/rstudio/rmarkdown/issues/701#issuecomment-237182996) to get the intermediaries files on local drive. (my home directory was on local drive). 

> although there is a very simple work around (store Rmd file on C:\ rather than network drive), some of us are not allowed to save files on our C:\ (per company policy and/or using RStudio Server Pro with mounted network folders)

I think any local folder where you have access right will work. Not just C:/ . You could have a home directory or even a temp directory to move your file before knitting. (R has one in `tempdir()`)
Moving files there before rendering could solve the issue with pandoc. Updating pandoc could help too maybe (`rmarkdown::pandoc_version()` to see what is shipped with your RStudio)

Hope it helps.

**Comment by vnijs** [comment]
Seeing the same issue using Rstudio in a windows VM. Setting the default working directory in Rstudio to a drive letter (e.g., Z:) rather than the default "~" resolved the issue.

**Comment by arielfuentes** [comment]
I'm trying to do multiple reports (https://www.reed.edu/data-at-reed/software/R/markdown_multiple_reports.html) and with mtcars works fine, but if I do the same with my own dataframe loaded from a *.csv doesn't work. I don't know how to fix it nor I understand why and how is happening, I'm on a local computer.

sessionInfo()
R version 3.5.0 (2018-04-23)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=Spanish_Spain.1252  LC_CTYPE=Spanish_Spain.1252    LC_MONETARY=Spanish_Spain.1252
[4] LC_NUMERIC=C                   LC_TIME=Spanish_Spain.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] readr_1.1.1    stringr_1.3.1  leaflet_2.0.2  dplyr_0.8.3    ggplot2_3.1.1  rmarkdown_1.10 markdown_0.8  
[8] knitr_1.20    

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.1       pillar_1.3.1     compiler_3.5.0   later_0.7.5      plyr_1.8.4       tools_3.5.0     
 [7] digest_0.6.18    evaluate_0.10.1  tibble_2.1.3     gtable_0.2.0     pkgconfig_2.0.2  rlang_0.4.0     
[13] cli_1.0.1        shiny_1.1.0      rstudioapi_0.7   crosstalk_1.0.0  yaml_2.1.19      withr_2.1.2     
[19] hms_0.4.2        htmlwidgets_1.3  rprojroot_1.3-2  grid_3.5.0       tidyselect_0.2.5 glue_1.3.1      
[25] R6_2.2.2         fansi_0.4.0      purrr_0.2.5      magrittr_1.5     promises_1.0.1   scales_1.0.0    
[31] backports_1.1.2  htmltools_0.3.6  rsconnect_0.8.15 assertthat_0.2.0 xtable_1.8-3     mime_0.5        
[37] colorspace_1.3-2 httpuv_1.4.5     utf8_1.1.4       stringi_1.2.2    lazyeval_0.2.1   munsell_0.5.0   
[43] crayon_1.3.4

**Comment by mymil** [comment]
I encountered the same error knitting to HTML from an .Rmd on a network drive. My working directory was already local. However, clearing the knitr cache resolved the issue.

**Comment by jacobgellman** [comment]
The solution by @vthivierge worked for me.

**Comment by knjpollard** [comment]
100% fix

when you open a project on a mapped drive do not open it through the shortcut to the mapped drive

#this works and happens automatically when you open the Rstudio project through that path
rmarkdown::render('H:/Mediasonic/John/tester.Rmd',  encoding = 'UTF-8');

this is what happens when you open it wrong the knit command cannot operate on this path
rmarkdown::render('//offfice//H//Mediasonic RAID1//John/tester.Rmd',  encoding = 'UTF-8');

**Comment by knjpollard** [comment]
Watch out used open file in recents and in "Quick Access" in windows 10 when opening Rstudio projects always brows to the actual mapped drive letter or set up a short cut that explicitly does that.

**Comment by vliublinska** [comment]
> Watch out used open file in recents and in "Quick Access" in windows 10 when opening Rstudio projects always brows to the actual mapped drive letter or set up a short cut that explicitly does that.

Thank you!! This solved it for me.

**Comment by vickibendus** [comment]
I'm having the same issue on a Mac. Would anyone be able to assist on how to map the files properly to overcome this glitch since there is no C:/ drive? It seems like the solution is possible but I can't get it to work on my system! Thank you!

**Comment by mymil** [comment]
You can use "~/" as a shortcut for home directory, or you can refer to the
absolute path, which will be something like /Users/Vicki , /Users/[your
username on the machine]

Joe Wasserman
joe.wasserman@gmail.com
https://www.joewasserman.com

**Comment by vickibendus** [comment]
Shoot that doesn't seem to be a solution for me then. My current working directory is "/Users/vickibendus/Desktop/R Materials/Soccer Reports", which appears to be the correct type of path. I'll have to keep searching for a way around this problem.

**Comment by haroldgil** [comment]
None of the solutions above have worked for me unfortunately. In my case, what confuses me is that the .Rmd file is converted to a .knit.md file which I cannot find (perhaps it's deleted since the rendering fails. My other question is why does pandoc get applied to a .utf8.md file if what was explicitly created was instead a .knit.md file. Perhaps a .utf8.md file is also created but I also cannot find this file (again it may just be deleted when rendering fails). 

![image](https://user-images.githubusercontent.com/6191338/72380904-639e8d00-36e4-11ea-9e2e-5f5a2ddb9e61.png)

I've taken the long pandoc command and have manually edited the paths to look more like C:/etc. This would work fine except that these .knit.md and .utf8.md files are not available so the error regarding openBinaryFile seems to be simply because it can't find these files (perhaps because they're automatically deleted but maybe not). Does anyone know where these files would temporarily exist at least?

**Comment by yihui** [comment]
@haroldgil To avoid deletion of the intermediate files, you may call `rmarkdown::render('matt.Rmd', clean = FALSE)`.

**Comment by haroldgil** [comment]
I found an easy solution in my case worth sharing. Again the problem seems to stem from the odd representation of a network drive directory. This directory comes into play in the pandoc command because pandoc() seems to indirectly rely on `Sys.getenv('R_USER')`.

![image](https://user-images.githubusercontent.com/6191338/72635810-b1590680-392b-11ea-9ff5-e9dd9f130fe2.png)

This directory is equivalent to my directory H:/
Thus I created a .Renviron file in H:/ and included in it the single entry R_USER=H:/
I restarted RStudio and the knitting works fine now. Also, looking at the pandoc command, all the weird references to the long/weird representation of H:/ (see image) have been replaced simply with H:/ indicating to me that pandoc() indeed relies on the value of `Sys.getenv('R_USER')`.

**Comment by RemKamal** [comment]
I tried all above and did not work.

I did know I had two paths in .libpaths() and it suspected it was causing the problem. It referred to one old R/RStudio on disk H, and, of course, it was ruining everything. 

How to navigate/find paths is explained here:
https://www.accelebrate.com/library/how-to-articles/r-rstudio-library
I just did this:
.libPaths(myPaths[2])  # reassign them

And then checked in "Install packages" that second path is missing.

After that KNIT complained about a bunch of missing packages, which my primary path in C did not have. After I installed it, it is working.

It is funny and sad at the same time, even though second path was not default, it was the root of the problem.

**Comment by jackhannah95** [comment]
> I found an easy solution in my case worth sharing. Again the problem seems to stem from the odd representation of a network drive directory. This directory comes into play in the pandoc command because pandoc() seems to indirectly rely on `Sys.getenv('R_USER')`.
> 
> ![image](https://user-images.githubusercontent.com/6191338/72635810-b1590680-392b-11ea-9ff5-e9dd9f130fe2.png)
> 
> This directory is equivalent to my directory H:/
> Thus I created a .Renviron file in H:/ and included in it the single entry R_USER=H:/
> I restarted RStudio and the knitting works fine now. Also, looking at the pandoc command, all the weird references to the long/weird representation of H:/ (see image) have been replaced simply with H:/ indicating to me that pandoc() indeed relies on the value of `Sys.getenv('R_USER')`.

I plumbed the depths of Google to find a fix for this problem (on Windows 10) and this was it. My `Sys.getenv("R_USER")` location was a network drive, as was the (different) location of my R project and the `.Rmd` file I was trying to knit. My steps to fix were:

1. [Mapped](https://support.microsoft.com/en-gb/help/4026635/windows-map-a-network-drive) the `Sys.getenv("R_USER")` location to my H:/ directory (pretty much any letter would've done).
2. Ran `usethis::edit_r_environ()` and added `R_USER=H:/` to it.
3. Restarted RStudio.
4. Re-ran `Sys.getenv("R_USER")` to ensure it was now `H:/`.
5. Mapped the folder containing the R project to my G:/ directory (again most letters would've done).
6. Opened the R project from my G:/ directory (either via the shortcut or by typing the path out; both worked for me) instead of the network drive.

Hope this is able to help someone else in future.

**Comment by geopang** [comment]
> > I found an easy solution in my case worth sharing. Again the problem seems to stem from the odd representation of a network drive directory. This directory comes into play in the pandoc command because pandoc() seems to indirectly rely on `Sys.getenv('R_USER')`.
> > ![image](https://user-images.githubusercontent.com/6191338/72635810-b1590680-392b-11ea-9ff5-e9dd9f130fe2.png)
> > This directory is equivalent to my directory H:/
> > Thus I created a .Renviron file in H:/ and included in it the single entry R_USER=H:/
> > I restarted RStudio and the knitting works fine now. Also, looking at the pandoc command, all the weird references to the long/weird representation of H:/ (see image) have been replaced simply with H:/ indicating to me that pandoc() indeed relies on the value of `Sys.getenv('R_USER')`.
> 
> I plumbed the depths of Google to find a fix for this problem (on Windows 10) and this was it. My `Sys.getenv("R_USER")` location was a network drive, as was the (different) location of my R project and the `.Rmd` file I was trying to knit. My steps to fix were:
> 
> 1. [Mapped](https://support.microsoft.com/en-gb/help/4026635/windows-map-a-network-drive) the `Sys.getenv("R_USER")` location to my H:/ directory (pretty much any letter would've done).
> 2. Ran `usethis::edit_r_environ()` and added `R_USER=H:/` to it.
> 3. Restarted RStudio.
> 4. Re-ran `Sys.getenv("R_USER")` to ensure it was now `H:/`.
> 5. Mapped the location of my R project to my G:/ directory (again most letters would've done).
> 6. Opened the R project from my G:/ directory (either via the shortcut or by typing the path out; both worked for me) instead of the network drive.
> 
> Hope this is able to help someone else in future.

@jackhannah95 @haroldgil  Looking at your responses, it appears that the steps you did would be the solution to my issue. On my end, everything works fine; however, when a third-party contractor executes the same scripts I'm using in a SharePoint site, he's getting the error. Before the error, it appears the script is looking at a [UNC](https://www.pcmag.com/encyclopedia/term/unc) path where the rmarkdown package is installed rather than a network drive.  In addition, I checked the contractor's libPaths() and it appears that it's showing as a UNC path  rather than the mapped network drive format. 

To confirm your scenarios, did converting the UNC path to a mapped network drive resolve your issues? Also, up to what UNC path subfolder did you map to the network drive? e.g., only up to the user name or up to the actual R subfolder?

**Comment by jackhannah95** [comment]
@geopang Yeah converting UNC paths to mapped network drives resolved my issues. They resolved the issues for me alone however - any other user trying to access my scripts on a shared network drive had to convert any UNC paths associated with their account to mapped network drives as well.

I mapped up to username level. It'd probably work at a lower level too but I think mapping the top level drive makes life easier.

**Comment by rogerjbos** [comment]
> > I found an easy solution in my case worth sharing. Again the problem seems to stem from the odd representation of a network drive directory. This directory comes into play in the pandoc command because pandoc() seems to indirectly rely on `Sys.getenv('R_USER')`.
> > ![image](https://user-images.githubusercontent.com/6191338/72635810-b1590680-392b-11ea-9ff5-e9dd9f130fe2.png)
> > This directory is equivalent to my directory H:/
> > Thus I created a .Renviron file in H:/ and included in it the single entry R_USER=H:/
> > I restarted RStudio and the knitting works fine now. Also, looking at the pandoc command, all the weird references to the long/weird representation of H:/ (see image) have been replaced simply with H:/ indicating to me that pandoc() indeed relies on the value of `Sys.getenv('R_USER')`.
> 
> I plumbed the depths of Google to find a fix for this problem (on Windows 10) and this was it. My `Sys.getenv("R_USER")` location was a network drive, as was the (different) location of my R project and the `.Rmd` file I was trying to knit. My steps to fix were:
> 
> 1. [Mapped](https://support.microsoft.com/en-gb/help/4026635/windows-map-a-network-drive) the `Sys.getenv("R_USER")` location to my H:/ directory (pretty much any letter would've done).
> 2. Ran `usethis::edit_r_environ()` and added `R_USER=H:/` to it.
> 3. Restarted RStudio.
> 4. Re-ran `Sys.getenv("R_USER")` to ensure it was now `H:/`.
> 5. Mapped the location of my R project to my G:/ directory (again most letters would've done).
> 6. Opened the R project from my G:/ directory (either via the shortcut or by typing the path out; both worked for me) instead of the network drive.
> 
> Hope this is able to help someone else in future.

This solution worked for me. Thanks @jackhannah95!

**Comment by gorgitko** [comment]
For those having problems with parallel `render()` (`pandoc: *.html: openFile: does not exist (No such file or directory)`, see my solution https://github.com/rstudio/rmarkdown/issues/1632#issuecomment-545824711

**Comment by chrtro1978** [comment]
This workaround works for Win 10 with network drive. 
Connect to the network drive using the file browser in Rstudio. Choose any folder.
Rstudio now has the necessary permission and everything knits perfectly.

R version 3.6.1 (2019-07-05)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows Server x64 (build 14393)

Matrix products: default

locale:
[1] LC_COLLATE=Danish_Denmark.1252  LC_CTYPE=Danish_Denmark.1252    LC_MONETARY=Danish_Denmark.1252 LC_NUMERIC=C                   
[5] LC_TIME=Danish_Denmark.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] compiler_3.6.1  htmltools_0.5.0 tools_3.6.1     yaml_2.2.1      rmarkdown_2.2   knitr_1.28      xfun_0.14       digest_0.6.25  
 [9] rlang_0.4.8     evaluate_0.14  
>

**Comment by cderv** [comment]
Hi every one, 

Pandoc has made some new fix regarding UNC path on Windows (https://github.com/jgm/pandoc/issues/5127). 
I wonder if this could be the fix for this issue we have with Pandoc and Rmarkdown.

is someone having the issue able to use Pandoc 2.11 (or above) and try to knit a document ? 

Pandoc 2.11+ is shipped with the RStudio 1.4 Preview is one of you is using it. That would be the easiest way to have access to this version as an R user. Otherwise, it is easy to install a release (https://github.com/jgm/pandoc/releases) but you would need to add it to PATH or use `rmarkdown::find_pandoc` before calling `rmarkdown::render`

Thank you !

**Comment by MTF-UMass** [comment]
@cderv I am quite new at this, but I installed pandoc 2.11, used `rmarkdown::find_pandoc` (found 2.11), called `rmarkdown::render`, and got the same error, unfortunately. I am using Windows 10.

The above-described workaround of mapping R_USER to a network drive does work.

**Comment by cderv** [comment]
@MTF-UMass thanks a lot ! That is really good to know - that means unfortunately Pandoc still have the issue. 
Can you paste the error you get with Pandoc 2.11 to see if anything has changed ?

**Comment by MTF-UMass** [comment]
@cderv The error I get is: `pandoc.exe: \\: openBinaryFile: does not exist (No such file or directory)
Error: pandoc document conversion failed with error 1
Execution halted`

pandoc is 2.11.2

**Comment by cderv** [comment]
Can you share the whole pandoc command, which should be printed in the Rmarkdown pane or console ? I would like to see which path fails exactly. Thanks

**Comment by MTF-UMass** [comment]
@cderv absolutely, sorry about that
`
"C:/Program Files/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS 03-params.utf8.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output 03-params.html --lua-filter "\\umaaffp\users\mtaftferguso\R\win-library\4.0\rmarkdown\rmarkdown\lua\pagebreak.lua" --lua-filter "\\umaaffp\users\mtaftferguso\R\win-library\4.0\rmarkdown\rmarkdown\lua\latex-div.lua" --email-obfuscation none --self-contained --standalone --section-divs --template "\\umaaffp\users\mtaftferguso\R\win-library\4.0\rmarkdown\rmd\h\default.html" --no-highlight --variable highlightjs=1 --variable "theme:bootstrap" --include-in-header "C:\Users\MTAFTF~1\AppData\Local\Temp\RtmpKomXKh\rmarkdown-str42503354e80.html" --mathjax --variable "mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" 
pandoc.exe: \\: openBinaryFile: does not exist (No such file or directory)
Error: pandoc document conversion failed with error 1
Execution halted
`

**Comment by cderv** [comment]
Thanks a lot for your help !

**Comment by Eyayaw** [comment]
I am having the same problem on mac. 

``` r
rmarkdown::render("Analysis/presentation/presentation.Rmd")

#> pandoc: Documentation/references.bib: openBinaryFile: does not exist (No such file or directory)
#> Error: pandoc document conversion failed with error 1

knitr::knit2pdf("Analysis/presentation/presentation.Rmd")
#> Error in tinytex::latexmk(basename(out), engine = compiler, ...) : 
#> The input file 'presentation.md' does not have the .tex extension 
```
<details>
<summary>
Session info
</summary>

─ Session info ─────────────────────────────────────────────────────────
 setting  value                       
 version  R version 4.0.2 (2020-06-22)
 os       macOS  10.16                
 system   x86_64, darwin17.0          
 ui       RStudio                     
 language (EN)                        
 collate  en_US.UTF-8                 
 ctype    en_US.UTF-8                 
 tz       Europe/Berlin               
 date     2020-12-16                  

─ Packages ─────────────────────────────────────────────────────────────
 package      * version    date       lib
 abind          1.4-5      2016-07-21 [1]
 assertthat     0.2.1      2019-03-21 [1]
 base64enc      0.1-3      2015-07-28 [1]
 callr          3.5.1      2020-10-13 [1]
 class          7.3-17     2020-04-26 [1]
 classInt       0.4-3      2020-04-07 [1]
 cli            2.2.0      2020-11-20 [1]
 clipr          0.7.1      2020-10-08 [1]
 codetools      0.2-18     2020-11-04 [1]
 colorspace     2.0-0      2020-11-11 [1]
 crayon         1.3.4      2017-09-16 [1]
 crosstalk      1.1.0.1    2020-03-13 [1]
 DBI            1.1.0      2019-12-15 [1]
 desc           1.2.0      2018-05-01 [1]
 devtools       2.3.2      2020-09-18 [1]
 dichromat      2.0-0      2013-01-24 [1]
 digest         0.6.27     2020-10-24 [1]
 dplyr        * 1.0.2      2020-08-18 [1]
 e1071          1.7-4      2020-10-14 [1]
 ellipsis       0.3.1      2020-05-15 [1]
 evaluate       0.14       2019-05-28 [1]
 fansi          0.4.1      2020-01-08 [1]
 farver         2.0.3      2020-01-16 [1]
 fs             1.5.0      2020-07-31 [1]
 generics       0.1.0      2020-10-31 [1]
 ggplot2      * 3.3.2      2020-06-19 [1]
 glue           1.4.2      2020-08-27 [1]
 gtable         0.3.0      2019-03-25 [1]
 highr          0.8        2019-03-20 [1]
 hms            0.5.3      2020-01-08 [1]
 htmltools      0.5.0      2020-06-16 [1]
 htmlwidgets    1.5.2      2020-10-03 [1]
 KernSmooth     2.23-18    2020-10-29 [1]
 knitr          1.30       2020-09-22 [1]
 labeling       0.4.2      2020-10-20 [1]
 lattice        0.20-41    2020-04-02 [1]
 leafem         0.1.3      2020-07-26 [1]
 leaflet        2.0.3      2019-11-16 [1]
 leafsync       0.1.0      2019-03-05 [1]
 lifecycle      0.2.0      2020-03-06 [1]
 lwgeom         0.2-5      2020-06-12 [1]
 magrittr       2.0.1      2020-11-17 [1]
 memoise        1.1.0.9000 2020-08-13 [1]
 munsell        0.5.0      2018-06-12 [1]
 pillar         1.4.7      2020-11-20 [1]
 pkgbuild       1.1.0      2020-07-13 [1]
 pkgconfig      2.0.3      2019-09-22 [1]
 pkgload        1.1.0      2020-05-29 [1]
 png            0.1-7      2013-12-03 [1]
 prettyunits    1.1.1      2020-01-24 [1]
 processx       3.4.4      2020-09-03 [1]
 ps             1.4.0      2020-10-07 [1]
 purrr          0.3.4      2020-04-17 [1]
 R6             2.5.0      2020-10-28 [1]
 raster         3.4-5      2020-11-14 [1]
 RColorBrewer   1.1-2      2014-12-07 [1]
 Rcpp           1.0.5.4    2020-11-13 [1]
 readr        * 1.4.0      2020-10-05 [1]
 remotes        2.2.0      2020-07-21 [1]
 reprex         0.3.0      2019-05-16 [1]
 rlang          0.4.9      2020-11-26 [1]
 rmarkdown      2.6.0001   2020-12-16 [1]
 rprojroot      2.0.2      2020-11-15 [1]
 rstudioapi     0.13       2020-11-12 [1]
 scales         1.1.1      2020-05-11 [1]
 sessioninfo    1.1.1      2018-11-05 [1]
 sf           * 0.9-6      2020-09-13 [1]
 sp             1.4-4      2020-10-07 [1]
 stars          0.4-3      2020-07-08 [1]
 stringi        1.5.3      2020-09-09 [1]
 stringr        1.4.0      2019-02-10 [1]
 testthat       3.0.0      2020-10-31 [1]
 tibble         3.0.4      2020-10-12 [1]
 tidyr        * 1.1.2      2020-08-27 [1]
 tidyselect     1.1.0      2020-05-11 [1]
 tinytex        0.28       2020-12-14 [1]
 tmap         * 3.2        2020-09-15 [1]
 tmaptools      3.1        2020-07-01 [1]
 units          0.6-7      2020-06-13 [1]
 usethis        2.0.0      2020-12-10 [1]
 vctrs          0.3.5      2020-11-17 [1]
 viridisLite    0.3.0      2018-02-01 [1]
 whisker        0.4        2019-08-28 [1]
 withr          2.3.0      2020-09-22 [1]
 xfun           0.19       2020-10-30 [1]
 XML            3.99-0.5   2020-07-23 [1]
 yaml           2.2.1      2020-02-01 [1]
 source                            
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 Github (hadley/memoise@34e8039)   
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 Github (RcppCore/Rcpp@a72a27c)    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 Github (rstudio/rmarkdown@80f14b2)
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.0)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.2)                    
 CRAN (R 4.0.0)                    

[1] /Library/Frameworks/R.framework/Versions/4.0/Resources/library

</details>

**Comment by cderv** [comment]
@Eyayaw I don't think this is the same issue. i.e an issue with network drive as the last and recent comments are showing.

Pandoc will throw this error: `openBinaryFile: does not exist (No such file or directory)` when a file needed for the rendering is not found - but sometimes it is an indicator of an issue in the source file. 

In your case, Pandoc is warning that the bib file referenced in your Rmd document is not found. You should check the `bibliography` field and the path of the file. Can you confirm that everything is ok on this side ? 

Also about `knitr:::knit2pdf`, I believe this is for Rnw or Rrst document. see `?knit2pdf`.

Thank you.

**Comment by Eyayaw** [comment]
> @Eyayaw I don't think this is the same issue. i.e an issue with network drive as the last and recent comments are showing.
> 
> Pandoc will throw this error: `openBinaryFile: does not exist (No such file or directory)` when a file needed for the rendering is not found - but sometimes it is an indicator of an issue in the source file.
> 
> In your case, Pandoc is warning that the bib file referenced in your Rmd document is not found. You should check the `bibliography` field and the path of the file. Can you confirm that everything is ok on this side ?
> 
> Also about `knitr:::knit2pdf`, I believe this is for Rnw or Rrst document. see `?knit2pdf`.
> 
> Thank you.

@cderv Thank you for the quick explanation. Yeah, it works when I copy the 'references.bib' file to the folder where the '.Rmd' file lives. However, I still find it puzzling why `rmarkdown/pandoc` does not find the `Documentation/references.bib` file. In fact, I set the `Evaluate Chunks in Directory option to Project` and I expect that to work. I am missing something here? Thank you.

**Comment by leonvictorlima** [comment]
Hi,

I am not sure if is the same issue, but when I execute a script in RStudio in Remote Session on Machine Learning Server to create a Rmarkdown (pdf) file I got an error.

=======

Error: 
<Error>
   ----------------------------------------------------------------
   success: FALSE
   url: http://172.16.13.70:12800/sessions/b7d365b2-f754-4a66-bb80-e02cb905b643/execute
   method: POST
   status_code: 500
   content: 
      Link: NULL
      Message: a3bdea4a-2d1f-47c4-a7a6-5ff9b903c924
      ExceptionType: NULL
   ----------------------------------------------------------------

Canceling execution...

Error: Execution stopped.
In addition: Warning messages:
1: replacing previous import ‘plotly::filter’ by ‘stats::filter’ when loading ‘EMSDashboard’ 
2: replacing previous import ‘dplyr::lag’ by ‘stats::lag’ when loading ‘EMSDashboard’ 
  |......................................................................| 100%
  ordinary text without R code

"D:/ProgramFiles/Microsoft/MLServer/PYTHON_SERVER/Scripts/pandoc" +RTS -K512m -RTS Presentation.utf8.md --to beamer --from markdown+autolink_bare_uris+tex_math_single_backslash --output File.tex --variable theme=Frankfurt --highlight-style tango --pdf-engine pdflatex --self-contained --include-in-header defaults.tex --include-in-header beamerthemeEpimed.sty 

==========
 

By the way, when the same code is executed inside the Server (server console) this works fine!

I noticed some differences in the execution and the Rmarkdown compilation exit, for example: The argument "processing file: File.RMD " is not showed in Local execution. 

                                                                           -- Execution from Server --

==================

|.....                                                                 |   7%
  ordinary text without R code

  |..........                                                            |  14%
label: unnamed-chunk-1 (with options) 
List of 1
 $ include: logi FALSE

  |...............                                                       |  21%
  ordinary text without R code

  |....................                                                  |  29%
label: unnamed-chunk-2 (with options) 
List of 1
 $ child: chr "General.Rmd"

  |.......................                                               |  33%
   inline R code fragments

  |...............................................                       |  67%
label: unnamed-chunk-8
  |......................................................................| 100%
   inline R code fragments


  |.........................                                             |  36%
  ordinary text without R code

  |..............................                                        |  43%
label: unnamed-chunk-3 (with options) 
List of 1
 $ child: chr "Admissions.Rmd"

....

===========================================

While in Local, we have this "processing file: file.Rmd" being showed.!


                                                                        --- Local ---

========================

processing file: Presentation.Rmd
  |....                                                  |   7%
  ordinary text without R code

  |........                                              |  14%
label: unnamed-chunk-1 (with options) 
List of 1
 $ include: logi FALSE

  |............                                          |  21%
  ordinary text without R code

  |...............                                       |  29%
label: unnamed-chunk-2 (with options) 
List of 1
 $ child: chr "General.Rmd"



processing file: ./General.Rmd
  |..................                                    |  33%
   inline R code fragments

  |....................................                  |  67%
label: unnamed-chunk-8

==============

Anyone could give me some help in this issue? 

Tks a Lot!

**Comment by cderv** [comment]
Hi @leonvictorlima, 

> I am not sure if is the same issue

it is always preferred to open a new issue that could link to this one instead of commenting in another issue because discussion will always be mixed up. 

It would be best to follow if you opened a new issue and use correct formatting at least for code block part. See issue guide: https://yihui.org/issue/

> By the way, when the same code is executed inside the Server (server console) this works fine!

It probably means that there is no issue with **rmarkdown** otherwise it would not work in R console I think. 

> when I execute a script in RStudio in Remote Session on Machine Learning Server 

It could come from something in RStudio IDE or the ML server. Are you using Pro version of those services ? You could ask there support if so to help find the issue. 

Anyway, **rmarkdown** issue board is not the best place for generic question. It is a place for issue found on the package, or feature request for the package. It would be better to ask in place like https://community.rstudio.com if you don't have access to paid support regarding your tools. 

Unfortunately, I don't have anything to test and look more into it. And I don't think this is the same issue the one discussed in this GH ticket.

**Comment by leonvictorlima** [comment]
Hello, @cderv ,

Thank you a lot for your comments. I will check it carefully in order to not mix this one and have a topic specific for it!

In addition, I am gonna ask support for them as well.

I agree with you when say this is not a directly package issue. Maybe it is something in the middle, I mean, among the execution on Server instance and my local.  But, I need more tests to confirm it.

Thank you, take care!

**Comment by jmarshallnz** [comment]
Just chiming in with my experiences with RStudio 1.4+ and pandoc 2.14.1. 2.14.1 has fixed the ".Rmd file located in the user roaming profile (i.e. UNC path) issue", so if RStudio updates to this version of pandoc then that is solved as far as I can tell.

However, there is a related problem: If the .libPaths() is on a roaming profile (UNC) location, and rmarkdown is installed into that location then you get basically the same error, presumably due to pandoc not loading the template files etc correctly. This is supposed to be fixed in pandoc 2.14.1:

https://github.com/jgm/pandoc/commit/26ed7fb4f9e95a00607ab9cfddc48267c717d24b

But it seems that it isn't completely fixed. Am not sure exactly which bit is failing, output is like this:

```
"C:/Program Files/RStudio/bin/pandoc/pandoc" +RTS -K512m -RTS blah.knit.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output blah.html --lua-filter "\\tur-homes2.massey.ac.nz\jcmarsha\My Documents\R\win-library\4.1\rmarkdown\rmarkdown\lua\pagebreak.lua" --lua-filter "\\tur-homes2.massey.ac.nz\jcmarsha\My Documents\R\win-library\4.1\rmarkdown\rmarkdown\lua\latex-div.lua" --self-contained --variable bs3=TRUE --standalone --section-divs --template "\\tur-homes2.massey.ac.nz\jcmarsha\My Documents\R\win-library\4.1\rmarkdown\rmd\h\default.html" --no-highlight --variable highlightjs=1 --variable theme=bootstrap --include-in-header "C:\Users\jcmarsha\AppData\Local\Temp\3\Rtmpya8M4d\rmarkdown-strd784b525dcd.html" --mathjax --variable "mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML" 
```

In addition, two warnings are generated, suggesting it might be fixable within rmarkdown maybe?

```
Warning messages:
1: In normalizePath(path.expand(path), winslash, mustWork) :
  path[1]="\\tur-homes2.massey.ac.nz": The specified path is invalid
2: In normalizePath(path.expand(path), winslash, mustWork) :
  path[1]="\\tur-homes2.massey.ac.nz": The specified path is invalid
```

Everything works fine, and the warnings are not generated if the UNC path is mapped to a drive and then R_LIBS_USER is set accordingly.

**Comment by kiwifb** [comment]
It seems like windows people have some kind of work around. But non-windows using drive over cifs are not so lucky.

I have seen a similar issue on jupyter because by default they tried to enforce atomic write which didn't work so well on remote file system. It kind of works the first time but after that you have some kind of ghost files preventing things from working. I got the following message from the r markdown window
```
Error in abs_path(input) : 
  The file 'simulating-the-Roy-Rubin-Model-final.Rmd' does not exist.
Calls: <Anonymous> -> setwd -> dirname -> abs_path
In addition: Warning messages:
1: In file.create(to[okay]) :
  cannot create file 'simulating-the-Roy-Rubin-Model-final.Rmd', reason 'No such file or directory'
2: In normalizePath(path, winslash = winslash, mustWork = mustWork) :
  path[1]="simulating-the-Roy-Rubin-Model-final.Rmd": No such file or directory
Execution halted
```
when I look at my file system I have
```
-rwx--x--x 1 frb15 domain users 817988 Jul 28 23:42  simulating-the-Roy-Rubin-Model-final.html*
-????????? ? ?     ?                 ?            ?  simulating-the-Roy-Rubin-Model-final.knit.md
-rwx--x--x 1 frb15 domain users  17041 Jul 28 23:41 'simulating-the-Roy-Rubin-Model final.Rmd'*
-????????? ? ?     ?                 ?            ?  simulating-the-Roy-Rubin-Model-final.Rmd
```
But I feel one the reason things don't work, is that temporary files are created in place instead of a proper temporary location, `/tmp` on unix and whatever it would be in windows.

**Comment by cderv** [comment]
@jmarshallnz @kiwifb Thanks a lot that is very helpful ! With your comment, I believe we have something to reproduce. It require to setup an environment with UNC path, so it won't be straightforward for me. 

Also, we still don't know for sure if this is Pandoc or rmarkdown. It is probably both. 

@jabranham the commit you linked to (jgm/pandoc@26ed7fb) is in a release version since Pandoc 2.11. When you say 
> 2.14.1 has fixed the ".Rmd file located in the user roaming profile (i.e. UNC path) issue"

Do you have another fix in Pandoc in mind ? FWIW, next RStudio IDE version should be shipped with 2.14.1

@kiwifb The error you get is clearly not from Pandoc and it different. If you can reproduce, can you share also the `traceback()` after the error ? What `abs_path()` does is calling `file.exists()`. With rmarkdown, everything happens by default relatively to the input path, included intermediates. I believe mainly because those intermediates needs to be found by Pandoc and we would need to deal with absolute path everywhere including in the generated md file. Note that some resources are created in a tmp folder like special HTML headers to include.

We'll try to make quicker progress with your information, thanks for sharing ! We definitely need to reproduce this in order to better dig into the issue with file managements and computation with network drive. 

Thanks again

**Comment by jmarshallnz** [comment]
@cderv Yes, you're right about that commit being in 2.11. Strange. There definitely is a difference in behaviour between pandoc 2.11 and 2.14.1 with RStudio 1.4. With v2.11 I can reproduce (everytime) the fault where the .Rmd is stored in a UNC path it fails to build. With 2.14.1 this is gone, except for the case where rmarkdown is also installed on a UNC path.

I can't really see anything obvious in pandoc other than that commit, so not sure what fixed (part of) the problem.

**Comment by cderv** [comment]
Then it is possible that something in RStudio IDE helped fix this issue. There was also non-pandoc issue with RStudio IDE and UNC path I believe. 

With what you say, it could mean that next version of RStudio IDE with Pandoc 2.14.1 should be better. 

We would need to look then more at the issue with rmarkdown package on network drive. For this, I am thinking we may need to copy the resources included into the package locally so that Pandoc does not need to access them through network drive (if that is the issue). Currently, some resource are passed to pandoc using absolute path to rmarkdown installation folder.

**Comment by kiwifb** [comment]
My setup is a rstudio server running on ubuntu 20.04 with some remote folders mounted via cifs. The previous error I think had some extra layers of indirections because the filename had space in it, which lead to extra normalization.
The first time it knits OK
```
output file: simulating-the-Roy-Rubin-Modelfinal.knit.md

/usr/lib/rstudio-server/bin/pandoc/pandoc +RTS -K512m -RTS simulating-the-Roy-Rubin-Modelfinal.knit.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output simulating-the-Roy-Rubin-Modelfinal.html --lua-filter /usr/lib/R/library/rmarkdown/rmarkdown/lua/pagebreak.lua --lua-filter /usr/lib/R/library/rmarkdown/rmarkdown/lua/latex-div.lua --self-contained --variable bs3=TRUE --standalone --section-divs --template /usr/lib/R/library/rmarkdown/rmd/h/default.html --highlight-style tango --variable theme=united --include-in-header /tmp/RtmpaGRfIM/rmarkdown-str14a4f44e57f07c.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' 

Output created: simulating-the-Roy-Rubin-Modelfinal.html
```
But the second time (or sometimes more time, it is a bit finicky) I now get
```
Error in file(con, "w") : cannot open the connection
Calls: <Anonymous> -> <Anonymous> -> write_utf8 -> writeLines -> file
In addition: Warning message:
In file(con, "w") :
  cannot open file 'simulating-the-Roy-Rubin-Modelfinal.knit.md': No such file or directory
Execution halted
```
I am doing everything from the rstudio interface at the moment, how do I run things to get to the backtrace you want. I can run R directly from the command line if needed.

**Comment by kiwifb** [comment]
Figured it out by running `rmarkdown::render` from rstudio's console.
```
5: file(con, "w") 
4: writeLines(enc2utf8(text), con, ..., useBytes = TRUE)
3: write_utf8(res, output)
2: knitr::knit(knit_input, knit_output, envir = envir, quiet = quiet) 
1: rmarkdown::render("/home/frb15/Pdrive/simulating-the-Roy-Rubin-Modelfinal.Rmd",         "html_document")
```

**Comment by cderv** [comment]
That is a bit disturbing to me that this is sometimes working, and sometimes not. 

Thanks for sharing the setup. I'll see if I can get my hand of such configuration.

**Comment by kiwifb** [comment]
I think there is a lot to be done around some remote file system. What personally gets my attention as a programer in other language and context is that the intermediary file (in `.knit.md`) shouldn't be created there. It is transient and disappear once you are done. On most OS there are special place for those in `/tmp` (or windows equivalent) or `~/.cache` or other special folders if you are more sophisticated. There is no absolute guarantee of locality but it is way cleaner than leaving your stuff  next to your input and output - unless you are debugging.

**Comment by cderv** [comment]
I understand you point, and I think what you describe applies to more standard temporary content. However, for the specific usage here, we need to consider the way Pandoc is working. For example, it is highly possible that `.knit.md` will contain references to resource relatively to your Rmd document. If the resulting md file is elsewhere, those resources linked in the md doc won't be located relatively to the source file anymore and Pandoc won't find them during the conversion to HTML. Some processing would need to happen to handle absolute vs relative path correctly. That is just an example. Also, **rmarkdown** does write to `tempdir()` already some files that does not have this type of concerns. 

We are a bit off topic here. You can open a new issue if you want to discuss some ways to improve that (Knitr + Pandoc + LaTeX interaction needs to be considered here). Thanks !

**Comment by kiwifb** [comment]
Ha, it just proves that I am completely ignorant of the context and that I don't know what I am talking about. I have been spending a few hours looking at the code earlier today and I couldn't really untangle the various abstractions. R is not the language I am the most familiar with. 

I have come to that conversation because I am supporting a lecturer for whom I have organised the particular setup. The cifs mount is to access our university shared file system for convenience. As it turns out it is not currently convenient to run a number of things including knitting directly on it :)

**Comment by cderv** [comment]
Thanks for chiming in then ! 

We really appreciate having also external contributors with different background to share views and thoughts on the ecosystem because it brings new perspective. If you still want to look deeper with your setup (and I think that would us), you could try without R and rmarkdown, using directly [Pandoc ](https://pandoc.org/MANUAL.html )to convert a document with different scenarios. For example : 
* Using Pandoc on your cifs mount to convert a doc
* Using Pandoc to convert from local storage but using a resource from external storage 

If this does not work well at this level, then it is why we have a hard time looking at it from our level 🤔

**Comment by cacsfre** [comment]
Just in case: I was getting this error with a `learnr` tutorial (`shiny_prerendered`). It was working locally but it kept throwing this error when running through docker.  I kept trying to figure what was wrong with the `Dockerfile` but later realized the problem was due to the `index.html` file being copied into the Docker version of the tutorial. The `index.html` file referenced a third file (`footer.html`) using the wrong location, which was being copied from the locally run tutorial.

Solution: don't track/copy `index.html` but let `rmarkdown::run` or `rmarkdown::render` generate it each time.

**Comment by EPriske** [comment]
> > Following @pauvasquezh, @NilsOle or @goldingn were you able to solve this?
> 
> I solved it by changing my RStudio global options and setting the R version to be the local C:\ drive and not R on the network drive.

This solved my problem with R Markdown. Thank you so much

**Comment by yihui** [comment]
For those hitting this specifically when running `rmarkdown::render()` **in parallel** (e.g. `registerDoParallel(cores > 1)`, `foreach %dopar%`, or a fork cluster) — as reported by @NilsOle and others above — this was the same root cause as #1632: forked workers share `tempdir()`, and each render's cleanup deleted sibling renders' temp files mid-conversion. Fixed in #2631 — please try the dev version:

```r
remotes::install_github('rstudio/rmarkdown')
```

Note this does **not** address the other cause discussed in this thread — `.Rmd` files or R installed on a **network/UNC drive** on Windows (the `Sys.getenv('R_USER')` / `.libPaths()` reports). That's a separate issue and remains open.
